lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -22,17 +22,29 @@ defmodule DynamicSupervisor do
 
   The options given in the child specification are documented in `start_link/1`.
 
-  Once the dynamic supervisor is running, we can start children
-  with `start_child/2`, which receives a child specification:
+  Once the dynamic supervisor is running, we can use it to start children
+  on demand. Given this `Counter` GenServer:
 
-      {:ok, agent1} = DynamicSupervisor.start_child(MyApp.DynamicSupervisor, {Agent, fn -> %{} end})
-      Agent.update(agent1, &Map.put(&1, :key, "value"))
-      Agent.get(agent1, & &1)
-      #=> %{key: "value"}
+      defmodule Counter do
+        use GenServer
 
-      {:ok, agent2} = DynamicSupervisor.start_child(MyApp.DynamicSupervisor, {Agent, fn -> %{} end})
-      Agent.get(agent2, & &1)
-      #=> %{}
+        def start_link(initial), do: GenServer.start_link(__MODULE__, initial)
+        def inc(pid), do: GenServer.call(pid, :inc)
+
+        def init(initial), do: {:ok, initial}
+        def handle_call(:inc, _, count), do: {:reply, count, count + 1}
```

Comment:
Looks good to me! I believe you used the single line definitions for conciseness but I would go with the multi-line one for clarify.